sources/web/datalab/polymer/components/table-inline-details/table-inline-details.ts

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+/**
+ * Table inline details pane element for Datalab.
+ * Displays information about the selected BigQuery table file inline in the
+ * file browser item list.
+ */
+class TableInlineDetailsElement extends Polymer.Element {
+
+  /**
+   * File whose details to show.
+   */
+  public file: BigQueryFile;
+
+  /**
+   * Id for table whose inline-details to show.
+   */
+  public tableId: string;
+
+  private _apiManager: ApiManager;
+  private _fileManager: FileManager;
+  private _rows: gapi.client.bigquery.TabledataRow[];
+  private _table: gapi.client.bigquery.Table | null;
+  private _busy = false;
+  private readonly DEFAULT_MODE = 'NULLABLE';
+
+  static get is() { return 'table-inline-details'; }
+
+  static get properties() {
+    return {
+      _rows: {
+        notify: true, // For unit tests
+        type: Object,
+        value: null,
+      },
+      _table: {
+        notify: true, // For unit tests
+        type: Object,
+        value: null,
+      },
+      file: {
+        observer: '_fileChanged',
+        type: Object,
+        value: {},
+      },
+      schemaFields: {
+        computed: '_computeSchemaFields(_table)',
+        type: Array,
+      },
+      tableId: {
+        observer: '_tableIdChanged',
+        type: String,
+        value: '',
+      },
+    };
+  }
+
+  constructor() {
+    super();
+
+    this._apiManager = ApiManagerFactory.getInstance();
+    this._fileManager = FileManagerFactory.getInstance();
+  }
+
+  _fileChanged() {
+    if (this.file && this.file.id) {
+      // TODO(jimmc) - move this into BigQueryFile?
+      const path = this.file.id.path;
+      const parts = path.split('/');
+      this.tableId = parts[0] + ':' + parts[1] + '.' + parts[2];
+    } else {
+      this.tableId = '';
+    }
+  }
+
+  _tableIdChanged() {
+    const matches = this.tableId.match(/^(.*):(.*)\.(.*)$/);
+    if (matches && matches.length === 4) { // The whole string is matched as first result
```

Comment:
If the above comment applies we can just pick these parts from the parsing in `_fileChanged`.